src/buckify.rs

```diff
@@ -289,23 +287,51 @@ fn generate_git_fetch(repo: &str, commit_hash: &str) -> Result<Rule> {
     }))
 }
 
-/// Extract the "serde-rs/serde" part of "https://github.com/serde-rs/serde.git".
-pub fn short_name_for_git_repo(repo: &str) -> Result<&str> {
-    static GITHUB_URL_REGEX: OnceCell<Regex> = OnceCell::new();
-    let github_url_regex = GITHUB_URL_REGEX
-        .get_or_try_init(|| Regex::new(r"^https://github.com/([[:alnum:].-]+/[[:alnum:].-]+)$"))?;
+/// Create a uniquely hashed directory name for the arbitrary source url
+pub fn short_name_for_git_repo(repo: &str) -> Result<String> {
+    anyhow::ensure!(
+        repo.starts_with("https://"),
+        "only https git urls are supported"
+    );
 
-    let repo_without_extension = repo.strip_suffix(".git").unwrap_or(repo);
-    if let Some(captures) = github_url_regex.captures(repo_without_extension) {
-        Ok(captures.get(1).unwrap().as_str())
-    } else {
-        // TODO: come up with some mangling scheme for arbitrary repo URLs. Buck
-        // does not permit ':' in target names.
-        bail!(
-            "unsupported git URL: {:?}, currently vendor=false mode only supports \"https://github.com/$OWNER/$REPO\" repositories",
-            repo,
-        );
+    // The strategy here is similar to what cargo does to generate a unique directory name
+    // for git sources
+    let mut sanitized = repo.to_lowercase();
+
+    if let Some(query) = sanitized.rfind('?') {
+        sanitized.truncate(query);
+    }
+
+    if let Some(hash) = sanitized.rfind('#') {
+        sanitized.truncate(hash);
+    }
+
+    if sanitized.ends_with(".git") {
+        sanitized.truncate(sanitized.len() - 4);
+    }
+
+    let mut dir_name = sanitized
+        .split('/')
+        .next_back()
+        .unwrap_or("_empty")
+        .to_owned();
+
+    #[allow(deprecated)]
+    let hash = {
+        use std::hash::{Hash, Hasher, SipHasher};
+        let mut hasher = SipHasher::new_with_keys(0, 0);
+        sanitized.hash(&mut hasher);
+        hasher.finish()
+    };
```

Comment:
Please use a supported hash library from crates.io.

Reply:
I chose fnv since it was already a dependency of this crate.